test/performance/commCompare/gbt/many-to-one.chpl

```diff
@@ -0,0 +1,98 @@
+use Time;
+use CommDiagnostics;
+
+const numWorkerNodes = numLocales - 1;
+config const numTasksPerNode = min reduce ([i in 1..numWorkerNodes]
+                                           Locales(i).maxTaskPar);
+
+config const numOps = 10**6;
+const numOpsPerTask = numOps / (numWorkerNodes * numTasksPerNode);
+
+config const printConfig = false;
+config const printTimings = false;
+config const printCommDiags = false;
+
+var x0: int;
+var x0Atomic: atomic int;
+
+enum op_t {
+  opNone,
+  opGet,
+  opPut,
+  opAMO,
+  opFastOn,
+  opOn
+};
+
+use op_t;
+
+config param op = opNone;
+
+proc main() {
+  if printConfig {
+    writeln(numWorkerNodes, ' worker nodes, ',
+            numTasksPerNode, ' worker tasks per node, ',
+            numOps, ' ', op, (if numOps == 1 then '' else 's'));
+  }
+
+  if printCommDiags {
+    resetCommDiagnostics();
+    startCommDiagnostics();
+  }
+
+  const startTime = getCurrentTime();
+
+  coforall locIdx in 1..numWorkerNodes with ( ref x0, ref x0Atomic ) {
+    on Locales(locIdx) {
+      coforall 1..numTasksPerNode with ( ref x0, ref x0Atomic ) {
+        for i in 1..numOpsPerTask {
+          select op {
+            when opNone {
+            }
+
+            when opGet {
+              if i == 1 then x0 = 0; // prevent compiler loop-hoisting GET
+              infiniteSink(x0);
+            }
+
+            when opPut {
+              x0 = infiniteSource();
```

Comment:
This violates the MCM, right? Are we ok with that (and will  weassume that the compiler will not optimize the puts away or anything?)

Reply:
Yes, it does violate the MCM.  We don't care about the result value as such, though; we're just concerned that the communication happen.  I don't know how to force that to happen without impacting the timing, though, possibly severely.

Reply:
We could and I don't think that would invalidate the test since we're actually trying to get at the comm performance on a per-node basis rather than a per-memory-location basis.  But just a `numWorkerNodes`-sized receiver array wouldn't make this obey the MCM.  We'd actually need a `(numWorkerNodes*numTasksPerNode)`-sized array.